kong/plugins/method-restriction/handler.lua

```diff
@@ -0,0 +1,30 @@
+local BasePlugin = require "kong.plugins.base_plugin"
+local responses = require "kong.tools.responses"
+
+local MethodRestrictionHandler = BasePlugin:extend()
+
+MethodRestrictionHandler.PRIORITY = 940
+
+local function checkMethod(conf, method)
+  for i = 1,#conf.methods do
```

Comment:
Let me first say that I'm not an expert in Lua performance. That said, I think if would be interesting to perform a quick test using Luajit to measure if this type of iterator is faster than something like a bitwise operator (http://lua-users.org/wiki/BitwiseOperators), or a lookup table (using `set[key] ~= nil`). If any of these options were more efficient in terms of performance, then you could use the nginx cache (Kong has a few wrappers on top of it) to store such lookup table or bit array.